falcon/asgi/app.py

```diff
@@ -279,7 +280,7 @@ def __init__(self, *args, request_type=Request, response_type=Response, **kwargs
 
         self.add_error_handler(WebSocketDisconnected, self._ws_disconnected_error_handler)
 
-    async def __call__(self, scope, receive, send):  # noqa: C901
+    async def _call_impl(self, scope, receive, send):  # noqa: C901
```

Comment:
Another alternative is simply overwriting this with `__call__ = _wrap_asgi_coroutine_func(__call__)`. Thoughts?

Reply:
why not just use _wrap_asgi_coroutine_func as a decorator?

Reply:
Doh, right :man_facepalming: I'll try that.
You can tell I was exploring completely different solutions first. :grimacing: